index.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-unused-vars */
+let books = [];
+const title = document.querySelector('.boot-title');
+const author = document.querySelector('.author');
+const btn = document.querySelector('.btn');
+const bookHolder = document.querySelector('#bookList');
+
+function addBook() {
+  const id = Math.random();
+
+  books.push({ id, title: title.value, author: author.value });
+  bookHolder.innerHTML += ` <span>${title.value}</span><br/>
+  <span>${author.value}</span><br/>
+  <button onclick="removeBook(${id})">Remove</button>
+  <hr>`;
+
+  title.value = '';
+  author.value = '';
+  localStorage.setItem('data', JSON.stringify(books));
+}
+
+btn.addEventListener('click', addBook);
+
+const removeBook = (id) => {
+  bookHolder.innerHTML = '';
+  books = books.filter((book) => book.id !== id);
+
+  for (let x = 0; x < books.length; x += 1) {
+    bookHolder.innerHTML += ` <span>${books[x].title}</span><br/>
+  <span>${books[x].author}</span><br/>
+  <button onclick="removeBook(${books[x].id})">Remove</button>
+  <hr>`;
+  }
+  localStorage.setItem('data', JSON.stringify(books));
+};
+window.onload = function onload() {
+  books = JSON.parse(localStorage.getItem('data'));
+
+  for (let x = 0; x < books.length; x += 1) {
+    bookHolder.innerHTML += ` <span>${books[x].title}</span><br/>
+  <span>${books[x].author}</span><br/>
+  <button onclick="removeBook(${books[x].id})">Remove</button>
+  <hr>`;
+  }
+};
```

Comment:
- [ ] As mentioned by the previous reviewer, your add method is not working from a different browser than yours, the reason for this is that when the window loads, you give a new value to your `books` array. this is where the problem is, as you can see on line **40** you are trying to get the data key from the localstorage 👉  `books = JSON.parse(localStorage.getItem('data'));` as you can see `localStorage.getItem('data')` will return null for other browsers because they don't have this key in their localstorage yet. so you should add a condition that should check if the key `data` exist in the localstorage first, if it exist we use it and otherwise we use an empty array, it should be something like this 👇 
books = JSON.parse(localStorage.getItem('data')) || [];

With this, you will never get undefined, and it should work in all browsers 😉